toolchain/sem_ir/typed_insts.h

```diff
@@ -795,6 +795,29 @@ struct FunctionTypeWithSelfType {
   InstId self_id;
 };
 
+// An overloaded C++ function declaration.
+struct OverloadedCppFunctionDecl {
+  static constexpr auto Kind =
+      InstKind::OverloadedCppFunctionDecl.Define<Parse::AnyFunctionDeclId>(
+          {.ir_name = "overloaded_cpp_fn_decl", .is_lowered = false});
```

Comment:
I'd expect that these values do get lowered, eg in cases such as:
```
import Cpp inline "void foo(); void foo(int);";
fn F() {
  Cpp.foo;
}
```
Since you lower `OverloadedCppFunctionType` as an empty struct type, you should lower this as an empty struct value (use `context.GetLiteralAsValue()`).
```suggestion
          {.ir_name = "overloaded_cpp_fn_decl"});
```

Reply:
I added a TODO for the moment, will take a look at it.